lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -1614,6 +1621,459 @@ class ConvertAtenAdaptivePoolOp : public OpConversionPattern<OpTy> {
+
+    indices[dimOffset] = high[0];
+    indices[dimOffset + 1] = low[1];
+    Value p10 = b.create<tensor::ExtractOp>(loc, input, indices);
+
+    indices[dimOffset] = high[0];
+    indices[dimOffset + 1] = high[1];
+    Value p11 = b.create<tensor::ExtractOp>(loc, input, indices);
+
+    // Let Aij := area rect((yProj,xProj) <-> (y_i*,x_j*)),
+    // where i* = i+1 mod 2 and x_0 = xLow, x_1 = xHigh etc.
+    // We interpolate via the weighted average of pij by weights Aij
+    // the formula is retval = Sum(pij*Aij for i and j in range(2)).
+    // Note: we do not need to divide by total rect area == 1.
+
+    // lengths : Aij == dyi*dxj
+    Value dy0 = b.create<arith::SubFOp>(loc, highFP[0], proj[0]);
+    Value dy1 = b.create<arith::SubFOp>(loc, proj[0], lowFP[0]);
+    Value dx0 = b.create<arith::SubFOp>(loc, highFP[1], proj[1]);
+    Value dx1 = b.create<arith::SubFOp>(loc, proj[1], lowFP[1]);
+
+    // left = A00*p00 + A01*p01 = dy0(dx0p00 + dx1p01)
+    Value dx0p00 = b.create<arith::MulFOp>(loc, dx0, p00);
+    Value dx1p01 = b.create<arith::MulFOp>(loc, dx1, p01);
+    Value sum = b.create<arith::AddFOp>(loc, dx0p00, dx1p01);
+    Value left = b.create<arith::MulFOp>(loc, dy0, sum);
+    // right = A10*p10 + A11*p11 = dy1(dx0p10 + dx1p11)
+
+    Value dx0p10 = b.create<arith::MulFOp>(loc, dx0, p10);
+    Value dx1p11 = b.create<arith::MulFOp>(loc, dx1, p11);
+    sum = b.create<arith::AddFOp>(loc, dx0p10, dx1p11);
+    Value right = b.create<arith::MulFOp>(loc, dy1, sum);
+
+    return b.create<arith::AddFOp>(loc, left, right);
+  }
+  LogicalResult
+  matchAndRewrite(Torch::TorchvisionRoiAlignOp op,
+                  typename Torch::TorchvisionRoiAlignOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op->getLoc();
+    Value result = op.getResult();
+
+    uint64_t samplingRatio =
+        cast<ConstantIntOp>(op.getSamplingRatio().getDefiningOp()).getValue();
+    int64_t samplingRatioInt = static_cast<int64_t>(samplingRatio);
+    Value pooledH = op.getPooledHeight();
+    Value pooledW = op.getPooledWidth();
+    Value spatialScaleVal = adaptor.getSpatialScale();
+    llvm::APFloat spatialScale =
+        cast<ConstantFloatOp>(op.getSpatialScale().getDefiningOp()).getValue();
+    Value rois = adaptor.getRois();
+    Value input = adaptor.getInput();
+    RankedTensorType inputType = dyn_cast_or_null<RankedTensorType>(
+        this->getTypeConverter()->convertType(op.getInput().getType()));
+
+    if (inputType == nullptr) {
+      op.emitError("Cannot determine input shape");
+    }
+
+    unsigned inputRank = inputType.getRank();
+    Value offset =
+        rewriter.create<arith::ConstantOp>(loc, rewriter.getF32FloatAttr(0.0));
+    RankedTensorType resultType = dyn_cast_or_null<RankedTensorType>(
+        this->getTypeConverter()->convertType(result.getType()));
```

Comment:
```suggestion
    auto resultType = getTypeConverter()->convertType<RankedTensorType>(result.getType());
```

Also, you should check that this conversion succeeded and emit an error otherwise